src/main/java/org/springframework/data/jdbc/repository/support/JdbcQueryLookupStrategy.java

```diff
@@ -27,33 +28,54 @@
 import org.springframework.data.repository.query.QueryLookupStrategy;
 import org.springframework.data.repository.query.RepositoryQuery;
 import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.SingleColumnRowMapper;
 
 /**
  * {@link QueryLookupStrategy} for JDBC repositories. Currently only supports annotated queries.
  *
  * @author Jens Schauder
+ * @author Kazuki Shimizu
  */
 class JdbcQueryLookupStrategy implements QueryLookupStrategy {
 
 	private final JdbcMappingContext context;
 	private final DataAccessStrategy accessStrategy;
+	private final ConversionService conversionService;
 
 	JdbcQueryLookupStrategy(EvaluationContextProvider evaluationContextProvider, JdbcMappingContext context,
 			DataAccessStrategy accessStrategy) {
 
 		this.context = context;
 		this.accessStrategy = accessStrategy;
+		this.conversionService = context.getConversions();
 	}
 
 	@Override
 	public RepositoryQuery resolveQuery(Method method, RepositoryMetadata repositoryMetadata,
 			ProjectionFactory projectionFactory, NamedQueries namedQueries) {
 
 		JdbcQueryMethod queryMethod = new JdbcQueryMethod(method, repositoryMetadata, projectionFactory);
-		Class<?> domainType = queryMethod.getReturnedObjectType();
-		RowMapper<?> rowMapper = new EntityRowMapper<>(context.getRequiredPersistentEntity(domainType),
-				context.getConversions(), context, accessStrategy);
-
+		Class<?> returnedObjectType = queryMethod.getReturnedObjectType();
+		RowMapper<?> rowMapper = context.getSimpleTypeHolder().isSimpleType(returnedObjectType)
+				? new CustomSingleColumnRowMapper<>(returnedObjectType)
```

Comment:
Can replace to the `SingleColumnRowMapper` that supports the ConversionService in spring-jdbc 5.0.4 ([SPR-16483](https://jira.spring.io/browse/SPR-16483).

Reply:
fix via b91da68c77a78c4ac44347a5e5fd1c54a6a80c9d